pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +515,87 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	log.Debugf("cluster flavor %q not recognised. Defaulting to false", c.clusterFlavor)
 	return false
 }
+
+// IsFakeAttachAllowed checks if the volume is eligible to be fake attached and returns a bool value
+func (c *K8sOrchestrator) IsFakeAttachAllowed(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) (bool, error) {
+	log := logger.GetLogger(ctx)
+	//Check pvc annotations
+	pvcAnn, err := c.getPVCAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get pvc annotations")
```

Comment:
Can you make the error message a bit more specific and add details? I have same comments in other places.